### PR TITLE
SCUMM: Also filter demo entries if detector reports multiple hits

### PR DESCRIPTION
**Pull request by tag2015**
When the gamedir contains multiple valid entries, narrow selection by platform and also demo status.
Fixes bug #10117 when using C64 images of both the full version and demo of Maniac Mansion in the same folder.

### CODE REVIEW

File: engines/scumm/metaengine.cpp [hunk 1]
```diff
@@ -299,6 +299,29 @@ Common::Error ScummMetaEngine::createInstance(OSystem *syst, Engine **engine) {
 		}
 	}
 
+	// Still no unique match found. Narrow down again excluding demos using extra
+	// metadata
+	if(results.size() > 1 && ConfMan.hasKey("extra")) {
```
Comment (bluegr) [inline engines/scumm/metaengine.cpp:304]:
```suggestion
	if (results.size() > 1 && ConfMan.hasKey("extra")) {
```

File: engines/scumm/metaengine.cpp [hunk 2]
```diff
@@ -299,6 +299,29 @@ Common::Error ScummMetaEngine::createInstance(OSystem *syst, Engine **engine) {
...
+		Common::String extra_cfg = ConfMan.get("extra");
+		Common::List<DetectorResult> tmp;
+
+		for (Common::List<DetectorResult>::iterator
+				  x = results.begin(); x != results.end(); ++x) {
+
+			//FIXME there's no demo flag in confman, this is not 100% reliable
```
Comment (bluegr) [inline engines/scumm/metaengine.cpp:311]:
```suggestion
			// FIXME: there's no demo flag in confman, this is not 100% reliable
```

File: engines/scumm/metaengine.cpp [hunk 3]
```diff
@@ -299,6 +299,29 @@ Common::Error ScummMetaEngine::createInstance(OSystem *syst, Engine **engine) {
...
+			if( (Common::String(x->extra).contains("Demo")) && (extra_cfg.contains("Demo")) )
```
Comment (bluegr) [inline engines/scumm/metaengine.cpp:312]:
```suggestion
			if (Common::String(x->extra).contains("Demo") && extra_cfg.contains("Demo"))
```

**Comment by tag2015** [comment]
Changes applied, sorry I duplicated the commit. Please squash

**Comment by bluegr** [comment]
Thanks for your work, looks good! Squashing

**Pull request merged**